ports/esp32/machine_wdt.c

```diff
@@ -42,20 +43,30 @@ typedef struct _machine_wdt_obj_t {
 STATIC machine_wdt_obj_t wdt_default = {{&machine_wdt_type}};
 
 STATIC mp_obj_t machine_wdt_make_new(const mp_obj_type_t *type_in, size_t n_args, size_t n_kw, const mp_obj_t *args) {
-    mp_arg_check_num(n_args, n_kw, 0, 1, false);
+    mp_arg_check_num(n_args, n_kw, 0, 2, true);
+    
+    enum {
+        ARG_timeout, ARG_panic
+    };
 
-    mp_int_t id = 0;
-    if (n_args > 0) {
-        id = mp_obj_get_int(args[0]);
-    }
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_timeout, MP_ARG_INT, {.u_int = 10} },
```

Comment:
You'll need to add a line for the `id`.  Check how ports/stm32/wdt.c does it.

Also, the default timeout should be 5 seconds to retain the original behaviour.

Reply:
What is the role of the id? The IDF documentation states that the function `esp_task_wdt_init()` only has timeout and panic parameters.
Thanks for your patience.

Reply:
It's so that it matches with other ports.  In some cases there may be multiple WDT's and the id will select which one.  It is an optional parameter so you can still do something like `machine.WDT(timeout=5000)`.